core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -149,7 +156,13 @@ public String defaultHostname() {
      * @return {@link Optional#empty()} if this {@link Server} did not start
      */
     public Optional<ServerPort> activePort() {
-        return Optional.ofNullable(Iterables.getFirst(unmodifiableActivePorts.values(), null));
+        if (isMutatingActivePorts) {
+            synchronized (activePorts) {
+                return Optional.ofNullable(Iterables.getFirst(activePorts.values(), null));
+            }
+        } else {
+            return Optional.ofNullable(Iterables.getFirst(activePorts.values(), null));
```

Comment:
This also has a similar issue. How about always acquiring a lock for simplicity and safety?